exauq/core/emulators.py

```diff
@@ -299,6 +300,12 @@ def fit(
             self._gp.theta
         )
 
+        if len(training_data) < (len(training_data[0].input) + 3):
```

Comment:
I'm not overly comfortable with the hardcoded + 3. It's not clear what parameters this is referring to. I'm guessing process variance, nugget and?...? This is a _classic magic number_ and should really be replaced with a named constant or computed value.  
```suggestion
        n_input_dims = len(training_data[0].input)
        # Estimate includes one lengthscale per input dimension, plus process variance and nugget
        n_params_estimated = n_input_dims + 3  

        if len(training_data) < n_params_estimated:
```

or... maybe we get the number of params from the gp(mogp)?:
`self._gp.n_params` [MOGP Docs](https://mogp-emulator.readthedocs.io/en/latest/implementation/GaussianProcess.html#mogp_emulator.GaussianProcess.GaussianProcess.n_params)

Reply:
Added check based on MOGP number of parameters. Definitely better, depends on covariance fn, how estimating nugget etc.